### PR TITLE
Do not retry deprovisioning if Kyma resource was not deleted

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

if Kyma resource was not deleted don't add the `Check_Kyma_Resource_Deleted` step as deprovision incomplete to avoid retrying the deprovision.

**Related issue(s)**

See #992

### CODE REVIEW

File: internal/process/deprovisioning/check_kyma_resource_deleted_step.go [hunk 1]
```diff
@@ -62,16 +62,15 @@ func (step *CheckKymaResourceDeletedStep) Run(operation internal.Operation, logg
 		Name:      kymaResourceName,
 	}, kymaUnstructured)
 
-	if err == nil {
-		logger.Infof("Kyma resource still exists")
-		return step.operationManager.RetryOperationWithoutFail(operation, step.Name(), "Kyma resource still exists", 5*time.Second, step.kymaResourceDeletionTimeout, logger, nil)
-	}
-
-	if !errors.IsNotFound(err) {
+	if !errors.IsNotFound(err) && err != nil {
```
Comment (jaroslaw-pieszka) [inline internal/process/deprovisioning/check_kyma_resource_deleted_step.go:65]: How isNotFound handles err = nil
Should not we change the order?
```suggestion
	if err != nil && !errors.IsNotFound(err) {
```

Comment (MarekMichali) [reply internal/process/deprovisioning/check_kyma_resource_deleted_step.go:65]: When err is nil then IsNotFound returns false.

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**